### DISCUSSION TITLE
When using groupby, change the type to distort the value.

### DISCUSSION THREAD

**Discussion by tommyhj217** [vaexio/vaex#2131] [Q&A]
Hi
DataFrame delay_time has values from -276 to 111 and dtype=int64.
![image](https://user-images.githubusercontent.com/86287388/180895338-8fc2bcb1-2405-43b1-8496-5a42386a0332.png)
The number of values is 206.
![image](https://user-images.githubusercontent.com/86287388/180896352-874b91d6-e21e-49d9-8f84-eb97f40666d4.png)

df_delaytime_data = vdf_2.groupby(['delay_time'], agg={'cnt':'count'})
![image](https://user-images.githubusercontent.com/86287388/180895537-c803424e-3c16-42af-88cc-ad5db21a7087.png)

However, if you do groupby, the dtype of the column is changed to int8, and the value of -128 or higher is changed to positive.

Is this a problem with groupby?
How can we solve it?

Thanks
Hyunjun

**Comment by JovanVeljanoski** [comment]
Hi, 

I can't reproduce your issue. Can you say anything more about the data, are you loading it from an hdf5 file, do you do any other operations before it etc..

This example works for me:

```python
import vaex
import numpy as np

x = np.arange(-218, 218, dtype=np.int64)
df = vaex.from_arrays(x=x)
df = df.export_hdf5('../tmp.hdf5')
df = vaex.open('../tmp.hdf5')
df.groupby(['x'], agg={'cnt': 'count'})
```

Can you provide a minimal example as above that showcases the issue?

**Reply by tommyhj217** [reply]
Hi 

Please check below capture and code.
It reproduce same issue.
![image](https://user-images.githubusercontent.com/86287388/180997677-01ef340e-74fa-4c10-9671-619562a4dc1f.png)

x = [-129,-120,15,30,40]
df = vaex.from_arrays(x=x)
df.groupby(['x'], agg={'cnt': 'count'})

Thanks

**Reply by maartenbreddels** [reply]
This seems like a bug, would be great if someone can open a PR with this in a test in tests/groupby_test.py, then I'm happy to fix it!

**Reply by JovanVeljanoski** [reply]
I confirm this is a bug and I can reproduce it via @tommyhj217 latest example! 
Thanks @tommyhj217 ! I will convert this discussion to an issue for better tracking.

**Reply by tommyhj217** [reply]
Thank you for checking and debugging issues.

As a result of checking more, if you change the type to float, it will not be reproduced.
I don't know, but is it related to the code below?
![image](https://user-images.githubusercontent.com/86287388/181122650-a1e6f94d-afa4-4e54-bcdf-7c8e3ec75751.png)

Thanks

**Reply by JovanVeljanoski** [reply]
No idea yet, but worth looking. Btw our core team just stared our holidays so we might not reply for the next 10-14 days or so :).
